db/migrate/20211117170147_set_tenant_id.rb

```diff
@@ -0,0 +1,17 @@
+class SetTenantId < ActiveRecord::Migration[5.0]
+  
+  def up
+    Account.where("provider = 1 AND master IS NULL").update_all(master: false)
+
+    {
+      BackendApi => :account,
+      LogEntry => :provider,
+      Alert => :account
+    }.each do |m, r|
+      m.joins(r).where(tenant_id: nil).where(accounts: { master: false }).each do |c|
+        c.update(tenant_id: c.send(r).tenant_id)
+      end
```

Comment:
Yes the same here about locked `accounts` table.
For DML (Data Manipulation Language) we prefer a rake task and if the data to migrate are too big, we enqueue a job. We then delete it later mainly in another release (because of on-prem)

Example:

https://github.com/3scale/porta/blob/master/lib/tasks/services.rake#L28-L31